### ISSUE TITLE
v2: Syndication UI

### ISSUE THREAD

**Issue by stephyiu** [Automattic/syndication#113]
From today's Syndication hack day, we agreed on some basic requirements for v2 UI that need to stay in place. All of these already exist in V1. This came out of a group conversation led by Tatyana G, who runs works on product at CBS.
- Syndication Settings (/wp-admin/options-general.php?page=push-syndicate-settings): The master site needs to be able to set what Post Types are syndicated, and a tick box to delete all the pushed posts when the master post is deleted.
- Edit Post Page: Has a metabox to select which groups of sites the post will get pushed to. Display top 5, and then click to show more. Option to check/uncheck all. 
- Add Site: oAuth2 authentication for WP API, and which Site Group a site gets added to

These items do not exist in v1, but are needed for V2:
- All Posts Page: For each post, show which group of sites a post was syndicated to. 
- Ability to set up email/Slack notifications for posts when they are syndicated

**Comment by joshbetz** [comment]
> Add Site: oAuth2 authentication for WP API, and which Site Group a site gets added to

WP-API core doesn't support oAuth2.

> These items do not exist in v1, but are needed for V2:

We should create separate issues for these and include them in a later version, I don't want to block v2 any longer than we need to.

**Comment by emrikol** [comment]
> WP-API core doesn't support oAuth2.

But I believe WordPress.com doesn't support oAuth1 (please correct me if I'm wrong).  So oAuth1 support will likely need to be in place in the "core" plugin, but we at VIP will likely need to take the burden of oAuth2 support for our platform.

**Comment by joshbetz** [comment]
Yep, just noting that we'll have to account for that somehow when we implement wp-api support.

**Comment by GaryJones** [comment]
> Ability to set up email/Slack notifications for posts when they are syndicated

While there was a notification system in the diverged/experimental 2.1 branch, I've opened #192 to capture its implementation for the 3.0.0 efforts.
